scripts/building/check_enforcer.ps1

```diff
@@ -0,0 +1,181 @@
+<#
+.DESCRIPTION
+Enforce the check of pipelines
+
+.EXAMPLE 
+PS> ./check_enforcer.ps1
+PS> ./check_enforcer.ps1 -MergeCommit <git_commit_sha> -LoopTimes 30
+
+#>
+[CmdletBinding()]
+param(
+    [string]$MergeCommit = "",
+    [int]$LoopTimes = 30 # Loop for 15 minutes at most.
+
+)
+
+$github_repository = 'microsoft/promptflow'
+$snippet_debug = 1 # Write debug info to console.
+
+
+if ($MergeCommit -eq "") {
+    git log -1 | ForEach-Object {
+        if ($_.Contains("Merge")) {
+            $MergeCommit = $_.Split(" ")[-3]
+        }
+    }
+    
+    if ($snippet_debug -eq 1) {
+        Write-Output "MergeCommit $MergeCommit"
+    }
+}
+
+function sdk_cli_trigger_checks([ref]$failed_reason_ref, [ref]$valid_status_array_ref) {
+    # Check if all sdk cli pipelines are triggered.
+    # return all valid checks when successful
+    $failed_reason_ref.Value = ""
+
+    $pipelines = @{
+        "sdk_cli_tests" = 4;
```

Comment:
I understand this (kind of) magic number correspond to the python versions? maybe add some comment here for the meaning of the value.

Reply:
OK, this is how many pipelines should I check against sdk_cli_tests?
There are 4.
I'll add comment